web/viewer.js

```diff
@@ -291,7 +291,8 @@ var PDFViewerApplication = {
       newScale = (newScale * DEFAULT_SCALE_DELTA).toFixed(2);
       newScale = Math.ceil(newScale * 10) / 10;
       newScale = Math.min(MAX_SCALE, newScale);
-    } while (--ticks && newScale < MAX_SCALE);
+      --ticks;
+    } while (ticks >= 0 && newScale < MAX_SCALE);
```

Comment:
I think that changing these lines to `} while (--ticks > 0 && newScale < MAX_SCALE);` should achieve what you want here (similar below).